packages/graphql-transformer-common/src/definition.ts

```diff
@@ -272,6 +274,29 @@ export function extendFieldWithDirectives(field: FieldDefinitionNode, directives
   return field;
 }
 
+export function extendObjectWithDirectives(object: ObjectTypeDefinitionNode, directives: DirectiveNode[]): ObjectTypeDefinitionNode {
+  if (!directives || directives.length === 0) {
+    return object;
+  }
+
+  const newDirectives = [];
+
+  for (const directive of directives) {
+    if (!object.directives.some((d) => d.name.value === directive.name.value)) {
```

Comment:
Optionally could use `hasDirectiveWithName` but it also matches the rest of the functions in this file as is.